spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/source/BaseReader.java

```diff
@@ -179,23 +179,27 @@ protected InputFile getInputFile(String location) {
 
   private Map<String, InputFile> inputFiles() {
     if (lazyInputFiles == null) {
-      Stream<EncryptedInputFile> encryptedFiles =
-          taskGroup.tasks().stream().flatMap(this::referencedFiles).map(this::toEncryptedInputFile);
-
-      // decrypt with the batch call to avoid multiple RPCs to a key server, if possible
-      Iterable<InputFile> decryptedFiles = table.encryption().decrypt(encryptedFiles::iterator);
+      Stream<InputFile> inputFiles =
+          taskGroup.tasks().stream().flatMap(this::referencedFiles).map(this::toInputFile);
```

Comment:
This should not change because it breaks usage of the bulk decrypt API. That doesn't matter for Iceberg standard encryption, but it would negatively affect people using their own `EncryptionManager`.

There's also no need to change how this works. It's fine for the `decrypt` method to return `BaseEncryptedInputFile.encryptedInputFile`. That doesn't actually create a decryption stream unless the `InputFile` is used.

Instead, we can either keep the `EncryptedInputFile` instances around or we can use a class that handles both APIs.

Reply:
I'll revert the changes in this class, and move the native decryption logic to the StandardEncryptionManager class.